azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ChildListFlattener.java

```diff
@@ -0,0 +1,181 @@
+package com.microsoft.azure.management.compute.implementation;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.Page;
+import com.microsoft.azure.PagedList;
+import com.microsoft.rest.RestException;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * {@link ChildListFlattener} that can take a paged list of parents and flatten their child lists
+ * as a single lazy paged list.
+ *
+ * @param <T> the type of parent paged list item
+ * @param <U> the type of child paged list item
+ */
+public final class ChildListFlattener<T, U> {
+    protected final String switchToCousin = "switchToCousin";
+    protected Iterator<T> parentItr;
+    protected PagedList<U> currentChildList;
+    protected PagedList<U> cousinList;
+    private final ChildListLoader<T, U> childListLoader;
+
+    /**
+     * Interface that will be implemented by the consumer of {@link ChildListFlattener}.
+     * <p>
+     * implementation will be used by {@link ChildListFlattener#flatten()} to load child
+     * paged list of parents in the parent paged list.
+     *
+     * @param <T> the parent type
+     * @param <U> the type of items in the child list
+     */
+    interface ChildListLoader<T, U> {
```

Comment:
it'd be helpful to readability to name the generic params more suggestively than T and U - for example: ParentT, ChildT